src/main/java/org/broadinstitute/dsm/jobs/TestBostonUPSTrackingJob.java

```diff
@@ -1,202 +1,211 @@
-            "WHERE dsm_kit_id <> 0 and  tracking_to_id = ? and dsm_kit_request_id in ( SELECT dsm_kit_request_id FROM ddp_kit_request where external_order_number = ? )";
-    private static final String SQL_UPDATE_UPS_RETURN_STATUS = "UPDATE ddp_kit SET ups_return_status = ?, ups_return_date = ? " +
-            "WHERE dsm_kit_id <> 0 and tracking_return_id= ? and dsm_kit_request_id in ( SELECT dsm_kit_request_id FROM ddp_kit_request where external_order_number = ? )";
-
-    private static final String SQL_SELECT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER = "select  eve.*,   request.ddp_participant_id,   request.ddp_label,   request.dsm_kit_request_id, request.ddp_kit_request_id, request.upload_reason, " +
-            "        realm.ddp_instance_id, realm.instance_name, realm.base_url, realm.auth0_token, realm.notification_recipients, realm.migrated_ddp, kit.receive_date, kit.scan_date" +
-            "        from ddp_kit_request request, ddp_kit kit, event_type eve, ddp_instance realm where request.dsm_kit_request_id = kit.dsm_kit_request_id and request.ddp_instance_id = realm.ddp_instance_id" +
-            "        and not exists " +
-            "                    (select 1 from EVENT_QUEUE q" +
-            "                    where q.DDP_INSTANCE_ID = realm.ddp_instance_id" +
-            "                    and " +
-            "                    q.EVENT_TYPE = eve.event_name" +
-            "                    and " +
-            "                    q.DSM_KIT_REQUEST_ID = request.dsm_kit_request_id " +
-            "                    and q.event_triggered = true" +
-            "                    )"+
-            "        and (eve.ddp_instance_id = request.ddp_instance_id and eve.kit_type_id = request.kit_type_id) and eve.event_type = ? "+
-            "         and realm.ddp_instance_id = ?" +
-            "          and kit.dsm_kit_request_id = ?";
-
-    static String DELIVERED = "DELIVERED";
-    static String RECEIVED = "RECEIVED";
-
-        private static String SELECT_BY_EXTERNAL_ORDER_NUMBER = "and request.external_order_number = ?";
-    private static Covid19OrderRegistrar orderRegistrar;
+
+    String DELIVERED = "DELIVERED";
+    String RECEIVED = "RECEIVED";
+
+    private String SELECT_BY_EXTERNAL_ORDER_NUMBER = "and request.external_order_number = ?";
+    private Covid19OrderRegistrar orderRegistrar;
 
     @Override
-    public void execute(JobExecutionContext context) throws JobExecutionException {
+    public void accept(PubsubMessage message, Context context) throws Exception {
+        if (message.data == null) {
+            logger.info("No message provided");
+            return;
+        }
+
+
+        logger.info(message.data + " *");
+        Config cfg = CFUtil.loadConfig();
+
+        String dbUrl = cfg.getString("dsmDBUrl");
+
+        PoolingDataSource<PoolableConnection> dataSource = CFUtil.createDataSource(20, dbUrl);
```

Comment:
Do we need 20 connections in the pool?  If this is single threaded, maybe we only need one?  It's important because if we proliferate CFs that each create so many connections, eventually we'll run out of them on the server side.